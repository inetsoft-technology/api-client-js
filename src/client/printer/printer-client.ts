/*
 * Copyright 2019 InetSoft Technology
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { AbstractClient } from "../abstract-client";
import { HttpAdapter } from "../http-adapter";
import { PrinterApi } from "./printer-api";

/**
 * @hidden
 */
export class PrinterClient extends AbstractClient implements PrinterApi {

   constructor(adapter: HttpAdapter<any>, token: () => Promise<string>, baseUrl: string) {
      super(adapter, token, baseUrl);
   }

   public getPrinters(): Promise<string[]> {
      const url = "api/public/printers";
      return this.doGet(url).then((list: any) => list.printers);
   }
}
