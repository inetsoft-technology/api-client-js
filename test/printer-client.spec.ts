/*
 * Copyright 2019 InetSoft Technology
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { PrinterClient } from "../src/client/printer/printer-client";

describe("PrinterClient", () => {
   let adapter: any;
   let client: PrinterClient;

   beforeEach(() => {
      adapter = {
         get: jest.fn(),
         post: jest.fn(),
         put: jest.fn(),
         delete: jest.fn(),
         map: jest.fn(),
         mapText: jest.fn(),
         mapBinary: jest.fn()
      };
      client = new PrinterClient(adapter, () => Promise.resolve("secret"), "http://localhost:8080/sree/");
   });

   describe("when getPrinters() is called", () => {
      it("then a list of printers is returned", () => {
         adapter.get.mockReturnValueOnce(Promise.resolve({ data: { printers: [ "printer1", "printer2", "printer 3" ]}}));
         adapter.map.mockImplementation((promise: Promise<any>) => promise.then(response => response.data));
         return client.getPrinters().then(result => {
            expect(result).toMatchSnapshot();
            expect(adapter.get.mock.calls.length).toBe(1);
            expect(adapter.get.mock.calls[0][0]).toBe("http://localhost:8080/sree/api/public/printers");
         });
      });
   });
});
