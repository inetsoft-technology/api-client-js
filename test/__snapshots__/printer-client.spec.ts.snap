// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrinterClient when getPrinters() is called then a list of printers is returned 1`] = `
Array [
  "printer1",
  "printer2",
  "printer 3",
]
`;
